xml/src/main/java/org/ehcache/xml/ConfigurationTranslator.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.xml;
+
+import org.ehcache.config.Configuration;
+import org.ehcache.config.ResourceType;
+import org.ehcache.spi.service.Service;
+import org.ehcache.spi.service.ServiceConfiguration;
+import org.ehcache.spi.service.ServiceCreationConfiguration;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+
+import javax.xml.XMLConstants;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerConfigurationException;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.ehcache.core.internal.util.ClassLoading.libraryServiceLoaderFor;
+
+public class ConfigurationTranslator {
+
+  static final String EHCACHE_NAMESPACE_PREFIX = "ehcache";
+  static final String CONFIG_ELEMENT_NAME = "config";
+  static final String SERVICE_ELEMENT_NAME = "service";
+  static final String COLON = ":";
+
+  private static final String INDENT_PROPERTY_KEY = "{http://xml.apache.org/xslt}indent-amount";
+  private static final String INDENT_SPACING = "2";
+  private static final String YES = "yes";
+
+  private static final URL CORE_SCHEMA_URL = XmlConfiguration.class.getResource("/ehcache-core.xsd");
+
+  private final DocumentBuilder domBuilder;
+  private final Configuration configuration;
+
+  private final Map<Class<? extends Service>, CacheManagerServiceConfigurationParser<?>> cacheManagerServiceConfigurationParserMap;
+  private final Map<Class<? extends Service>, CacheServiceConfigurationParser<?>> cacheServiceConfigurationParserMap;
+  private final Map<ResourceType<?>, CacheResourceConfigurationParser> cacheResourceConfigurationParserMap;
+
+  public ConfigurationTranslator(Configuration configuration) throws IOException, SAXException, ParserConfigurationException {
```

Comment:
I''m confused about this class... this PR add xml generation for a subset of the full xml configuration prior to the calling code being ready.  In that sense this class is going to be wholly replaced when Albin's code is merged?  In an ideal world those two PRs would have arrived in the opposite order to how we're getting them.  In this reality wouldn't it make more sense to either:

1. Have this PR be raised against Albin's branch.
2. Leave this class out of this PR so that it's 'correct' but detached, rather than using this class to 'attach' it, but in an invalid way.

Reply:
This class was required to do some flow testing. Myself and Albin were working simultaneously. As I already wrote a big part of it, we decided to push it in my PR. Albin could add functionalities on top of it. Anyways, as you said, I will ask Albin to put this change in his PR